apps/trustlab/src/payload/access/abilities.js

```diff
@@ -0,0 +1,67 @@
+import { checkRole } from "./checkRole";
+import { ROLE_ADMIN, ROLE_AUTHOR, ROLE_DEFAULT, ROLE_EDITOR } from "./roles";
+
+export const canManageContent = (user) => {
+  if (!user) return false;
+  const isAdmin = checkRole([ROLE_ADMIN], user);
+  if (isAdmin) return true;
+
+  const isEditor = checkRole([ROLE_EDITOR], user);
+  if (isEditor)
+    return {
+      or: [
+        {
+          createdBy: {
+            equals: user.id,
+          },
+        },
+        {
+          "createdBy.role": {
+            in: [ROLE_EDITOR, ROLE_AUTHOR],
+          },
+        },
+      ],
+    };
+
+  return {
+    createdBy: {
+      equals: user.id,
+    },
+  };
+};
+
+export const canManagePages = (user) =>
+  checkRole([ROLE_ADMIN, ROLE_EDITOR], user);
+
+export const canManageSiteSettings = (user) => checkRole([ROLE_ADMIN], user);
+
+export const canManageUsers = (user) => {
+  if (!user) return false;
+
+  if (user.role === ROLE_ADMIN) return true;
+
+  if (user.role === ROLE_DEFAULT)
```

Comment:
If manage = edit/remove, then:

1. Admins can `create` and `manage` all users
2. Editors can `manage` their own account and `create`/`manage` authors (an editor can not create or delete other editors or admins)
3. Authors can only `manage` their own account.

---

One thing to think about is what happens to the content when a user is removed.